Architecture detection with runtime.GOARCH

palfrey opened the pull request:
This PR replaces most of the instances of hardcoded "amd64" with runtime.GOARCH, so at least native builds work better on non-amd64 systems.

stapelberg commented:
Replacing hard-coded amd64 with runtime.GOARCH is certainly an improvement, but can you leave the TODOs in place please? This needs to be configurable, so the TODOs are still relevant.

Thanks,

palfrey commented:
> Replacing hard-coded amd64 with runtime.GOARCH is certainly an improvement, but can you leave the TODOs in place please? This needs to be configurable, so the TODOs are still relevant.

All the places where runtime.GOARCH was being just used without any checks, I've replaced/added TODOs. I've removed it in cases where a cross variable was being checked first. Does that make sense?

stapelberg commented:
Thanks!
